### ISSUE TITLE
@typescript-eslint/no-unused-vars should follow the same pattern with tsc by default

### ISSUE THREAD

**Issue by ShuiRuTian** [iamturns/eslint-config-airbnb-typescript#328]
```ts
function foo(a, _b){
}
```

For `foo`, tsc will treat `_b` as used, which means, if you enable config `noUnusedParameters`, there will be an error on `a`, but no error on `_b`.

It might be a good choice to follow the same pattern by default.
